test/integration/pkcs-tpm-ecc-support-check.int.c

```diff
@@ -0,0 +1,147 @@
+#include <openssl/evp.h>
+#include <openssl/rsa.h>
+#include <openssl/err.h>
+#include "test.h"
+
+
+struct test_info {
+    CK_SESSION_HANDLE handle;
+    CK_SLOT_ID slot_id;
+};
+
+static test_info *test_info_new(void) {
+
+    test_info *ti = calloc(1, sizeof(*ti));
+    assert_non_null(ti);
+
+    /* get the slots */
+    CK_SLOT_ID slots[6];
+    CK_ULONG count = ARRAY_LEN(slots);
+    CK_RV rv = C_GetSlotList(true, slots, &count);
+    assert_int_equal(rv, CKR_OK);
+
+    ti->slot_id = slots[0];
+
+    return ti;
+}
+
+static int test_setup(void **state) {
+
+    test_info *ti = test_info_new();
+
+    CK_RV rv = C_OpenSession(ti->slot_id, CKF_SERIAL_SESSION | CKF_RW_SESSION,
+            NULL, NULL, &ti->handle);
+    assert_int_equal(rv, CKR_OK);
+
+    *state = ti;
+
+    return 0;
+}
+
+static int test_teardown(void **state) {
+
+    test_info *ti = test_info_from_state(state);
+
+    CK_RV rv = C_CloseAllSessions(ti->slot_id);
+    assert_int_equal(rv, CKR_OK);
+
+    free(ti);
+
+    return 0;
+}
+
+static void test_tpm_ecc_support_check(void **state) {
+
+    test_info *ti = test_info_from_state(state);
+    CK_SESSION_HANDLE session = ti->handle;
+
+    CK_BBOOL ck_true = CK_TRUE;
+    CK_BBOOL ck_false = CK_TRUE;
+    CK_BYTE id[] = "p11-templ-key-id-ecc";
+    CK_UTF8CHAR label[] = "p11-templ-key-label-ecc";
+    CK_RV rv = CKR_GENERAL_ERROR;
+    CK_RV rv_ecc = CKR_GENERAL_ERROR;
+
+    CK_BYTE ec_params_P256[] = {
+        0x06, 0x08, 0x2a, 0x86, 0x48,
+        0xce, 0x3d, 0x03, 0x01, 0x07
+    };
+
+    CK_BYTE ec_params_P521[] = {
+        0x06, 0x05, 0x2b, 0x81, 0x04,
+        0x00, 0x23
+    };
+
+    /*
+     * Check the return values for supported and not supported curve.
+     * The first rv_ecc should be CKR_OK for supported P256 curve
+     * The second rv_ecc should be CKR_ARGUMENTS_BAD for unsupported P521 curve
+     */
+
+    CK_BYTE_PTR ec_params[2] ={ec_params_P256, ec_params_P521}; /*P256 is supported, P521 is not supported by simulator*/
+
+    user_login(session);
+
+    CK_OBJECT_HANDLE pubkey;
+    CK_OBJECT_HANDLE privkey;
+    CK_MECHANISM mech = {
+        .mechanism = CKM_EC_KEY_PAIR_GEN,
+        .pParameter = NULL,
+        .ulParameterLen = 0
+    };
```

Comment:
add empty line here, or remove empty line below

Reply:
I don't get, where exactly insert the line.

Reply:
du erstellst 3 strukturen hier mechs, priv und pub.
zwischen mech und priv hast du keine leere zeile
zwischen priv und pub hast du eine leere zeile.

Entweder zwischen mech und priv eine leere zeile hinzufügen oder die leere zeile zwischen priv und pub entfernen